superset/views/core.py

```diff
@@ -1560,8 +1560,16 @@ def tables(self, db_id, schema, substr, force_refresh='false'):
             max_tables = max_items * len(table_names) // total_items
             max_views = max_items * len(view_names) // total_items
 
-        table_options = [{'value': tn, 'label': tn}
-                         for tn in table_names[:max_tables]]
+        # Make modifications for Drill
+        # This hack (and it's a total hack) replaces the dot from the file extension
+        # with '@@@' to prevent Superset from misinterpreting the schema
+        if "drill+sadrill" in database.sqlalchemy_uri_decrypted:
```

Comment:
To avoid having `if database.engine == 'some_engine'` all over the codebase, we make sure to keep all of the database engine specific code in `db_engine_specs.py`. We may need to add some new hook in `db_engine_specs.py` to support your use case, though hopefully not a hack...

Reply:
@mistercrunch 
Perhaps hack was too strong a word, but I couldn't find an appropriate hook. 
The drop down is now populated as follows:
<option value="filename@@@csv">filename.csv</option>

Do you have any suggestions as to either how to write a hook to properly encode/decode the URL or otherwise get the arguments passed correctly?

Reply:
I mean you have to introduce a new hook to `db_engine_spec.BaseEngineSpec`, so that all the oddness of this hack (or whatever you'll call it) is limited to the `DrillEngineSpec` class. Something like `db_engine_spec.alter_table_name(table_name)`

Reply:
The goal is to not have any database-engine-specific logic outside of `db_engines_spec.py`, so that implementing a new engine is limited to creating a new class in that module, instead of altering methods scattered all around the code base.

Reply:
Fixed/